enzyme/Enzyme/ActivityAnalysisPrinter.cpp

```diff
@@ -90,6 +90,15 @@ bool printActivityAnalysis(llvm::Function &F, TargetLibraryInfo &TLI) {
       dt = ConcreteType(a.getType()->getScalarType());
     } else if (a.getType()->isPointerTy()) {
 #if LLVM_VERSION_MAJOR >= 17
+#elif LLVM_VERSION_MAJOR >= 13
+      if (!a.getType()->isOpaquePointerTy()) {
+        auto et = a.getType()->getPointerElementType();
+        if (et->isFPOrFPVectorTy()) {
+          dt = TypeTree(ConcreteType(et->getScalarType())).Only(-1, nullptr);
+        } else if (et->isPointerTy()) {
+          dt = TypeTree(ConcreteType(BaseType::Pointer)).Only(-1, nullptr);
+        }
+      }
 #else
       auto et = a.getType()->getPointerElementType();
```

Comment:
Can you change this to an if supportsTypedPointers in the else branch, like here? https://github.com/EnzymeAD/Enzyme/blob/12030cd074a6a167807de8eff733ee9b67e2a42f/enzyme/Enzyme/DiffeGradientUtils.cpp#L187